packages/cli-platform-ios/src/commands/buildIOS/index.ts

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import path from 'path';
+import chalk from 'chalk';
+import {Config} from '@react-native-community/cli-types';
+import {
+  logger,
+  CLIError,
+  getDefaultUserTerminal,
+} from '@react-native-community/cli-tools';
+import {Device} from '../../types';
+import {buildProject} from './buildProject';
+import {getDestinationSimulator} from '../../tools/getDestinationSimulator';
+import {getDevices} from '../../tools/getDevices';
+
+type FlagsT = {
+  configuration: string;
+  simulator?: string;
+  scheme?: string;
+  device?: string | true;
+  udid?: string;
```

Comment:
I've defaulted the destination to generic I don't think we should limit users from selecting a target.
I believe It can be commonly used as a way of building time optimization.